src/Pact/Analyze/Eval/Prop.hs

```diff
@@ -38,6 +38,9 @@ import qualified Pact.Analyze.Types         as Types
 import           Pact.Analyze.Types.Eval
 import           Pact.Analyze.Util
 
+#if ! MIN_VERSION_base(4,13,0)
+import Control.Monad.Fail (MonadFail(..))
+#endif
```

Comment:
`base-4.13` (ghc 8.8) exports `MonadFail` from Prelude. Before that it requires a special import.

Reply:
The true solution is that Pact should migrate away from usage of `fail` to the extent that it currently is.

Reply:
Why is MonadFail so special? I import Data.List all the time without a problem.

Reply:
I double-checked. By 8.8 [the MonadFail Proposal](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/8.8.1-notes.html) is complete. We had already accounted for the removal of `fail` from Monad some time ago, but with 8.8 the explicit import is no longer needed.

And due to Pact using `-Werror`, removing the CPP we see:
```
/home/colin/code/haskell/pact/src/Pact/Analyze/Eval/Prop.hs:41:1: error: [-Wunused-imports, -Werror=unused-imports]
    The import of ‘Control.Monad.Fail’ is redundant
      except perhaps to import instances from ‘Control.Monad.Fail’
    To import instances alone, use: import Control.Monad.Fail()
   |        
41 | import Control.Monad.Fail (MonadFail(..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
but only when compiling with 8.8.

Reply:
An alternative to CPP is to do ghc-8.8 builds with `-Wwarn`, which is what I do.

Reply:
The best, solution, however, would be to stop using `fail`.

Most of the times, there is a better, more specific, way of error handling available. We removed already most uses of it from the code. There are only very few left.